### DISCUSSION TITLE
Why some Model doesn't implement `Init()` method?

### DISCUSSION THREAD

**Discussion by RadhiFadlillah** [charmbracelet/bubbles#563] [Q&A]
From what I understand, every custom model for BubbleTea must implement `tea.Model` interface for composability. However,  from 15 models in this Bubbles package only 5 of them that implement `Init()` method. I thought the least we must do is creating an empty init function like how `Progress.Model` do:

```go
// Init exists to satisfy the tea.Model interface.
func (m Model) Init() tea.Cmd {
	return nil
}
```

Personally I prefer if every model implements `Init()` method, at least for consistency. Are there any reasons on why some models don't implement it?

Thanks!

**Comment by activey** [comment]
As these are not real models but a "components" with specific init methods for each. And yes, this is misleading and makes the whole architecture a bit wacky...
